### ISSUE TITLE
Parse time form yokogawa metadata

### ISSUE THREAD

**Issue by jluethi** [fractal-analytics-platform/fractal#132]
Some multiplexing approaches need to get the acquisition time as part of the metadata (for example, to correct for time-based intensity degradation to to multiplexing buffers). With 2b3aea8a980bd10a86cdd6b6dd96d22445a1ace3, I added the parsing and aggregation of time information per site to the yokogawa metadata parsing.

During aggregation, this parsing takes the first time point per field of view. If there are 3 channels and each has 20 Z planes, there are 60 different time stamps. Because we save metadata per field of view, this needs to be aggregated. My approach here is using the first time point per field of view across all the channels (by taking the `min` of all timepoints per FOV).

For each site, the parsed metadata table now looks like this:
<img width="1135" alt="20220803_metadata_table" src="https://user-images.githubusercontent.com/18033446/182591157-7caa225e-0c1b-4166-b479-6189ad77b340.png">

I would say such time information is optional metadata for the FOV ROI tables. @tcompa How do we best parse such optional metadata to the ROI tables without enforcing them to always be there (e.g. a user that provides their own table (https://github.com/fractal-analytics-platform/fractal/issues/124) may not have time information, if we just infer the table based on filenames (https://github.com/fractal-analytics-platform/fractal/issues/127), we certainly won't have this metadata)? (not urgent, more a September task)

@MaksHess: Is this sufficient for your use cases? I used this approach in the past to calculate intensity bias decay profiles.
Also, just FYI: I convert all times to UTC. The yokogawa time stamps are a bit annoying (e.g. default `datetime.strptime ` can't handle those time zones, pandas auto-detection gets weird time zone readouts unless we force a time zone). Now it gets correctly converted to UTC (which is 1-2 hours of our local time zone). Any bias correction would not be influenced by this anyway

**Comment by MaksHess** [comment]
I think one timestamp per site should suffice for now! :)
Regarding the time parsing, there's `datetime.datetime.fromisoformat` that parses the strings directly into `datetime.datetime` objects.

**Comment by jluethi** [comment]
@MaksHess Thanks! That's also where I started looking. Unfortunately, the yokogawa date strings aren't isoformat only, but contain timezones in a way that isn't supported by the datetime package (adding +02:00 at the end of the string), as far as I can tell. That's why I switched to pandas datetime parsing, which is a bit more flexible.
See here for further details: https://stackoverflow.com/questions/50489198/strptime-unable-to-resolve-timezone-value-0200
(alternative would be to chop off timezone info and just ignore it, but pandas actually parses it correctly)

**Comment by jluethi** [comment]
The additional time metadata broke some downstream parsing, so I already fixed it now. The AnnData table now contains time in the .obs part. It's a pandas table with a time column.

Time is saved as a iso string (e.g. `"2020-08-12 15:36:36.234000+0000"`), because the AnnData Zarr writer doesn't support writing `pandas.core.arrays.datetimes.DatetimeArray`. So for the discussion above: Whenever a user reads the string time, they can then convert it to whatever suits them best again :)

If we want to add further metadata in the future, the `prepare_FOV_ROI_table` is now generalized. It takes a list of metadata columns and saves all of them to .obs table in AnnData. It's also more stable now and doesn't crash anymore when extra columns are provided in the input dataframe (e.g. if we decide to parse additional things from another microscope or if a user provides extra metadata in a custom table).

Finally, @tcompa , I changed the FOV naming. Instead of assigning them numbers from 1 to n (e.g. FOV1 to FOV4), they now get the numbers that were their FOV index (still 1-4 in the simple example). The name string is now formatted as FOV_1 to FOV_4. This seems to pass all tests. The reason for this change: We may have datasets where we have e.g. FOVs 1, 2, 4, 5, 7. Would be nice that those indices are maintained as names of the FOVs. This change passes all our current unit tests and also the `test_apply_workflow_on_fake_data.py` integration test. Let me know if there could be issues from it though.
